opentelemetry-exporter-geneva/geneva-uploader/src/payload_encoder/otlp_encoder.rs

```diff
@@ -303,30 +303,25 @@ impl OtlpEncoder {
         &self,
         buffer: &mut Vec<u8>,
         attr: &opentelemetry_proto::tonic::common::v1::KeyValue,
-        expected_type: u8,
+        expected_type: BondDataType,
     ) {
-        const BT_STRING: u8 = BondDataType::BT_STRING as u8;
-        const BT_FLOAT: u8 = BondDataType::BT_FLOAT as u8;
-        const BT_DOUBLE: u8 = BondDataType::BT_DOUBLE as u8;
-        const BT_INT32: u8 = BondDataType::BT_INT32 as u8;
-        const BT_WSTRING: u8 = BondDataType::BT_WSTRING as u8;
-
         if let Some(val) = &attr.value {
             match (&val.value, expected_type) {
-                (Some(Value::StringValue(s)), BT_STRING) => BondWriter::write_string(buffer, s),
-                (Some(Value::StringValue(s)), BT_WSTRING) => BondWriter::write_wstring(buffer, s),
-                (Some(Value::IntValue(i)), BT_INT32) => {
-                    // TODO - handle i64 properly, for now we cast to i32
-                    BondWriter::write_int32(buffer, *i as i32)
+                (Some(Value::StringValue(s)), BondDataType::BT_STRING) => {
+                    BondWriter::write_string(buffer, s)
+                }
+                (Some(Value::StringValue(s)), BondDataType::BT_WSTRING) => {
```

Comment:
Do we expect to serialize an attribute as `wstring`?

Reply:
Good catch. Removed.